src/tsickle.ts

```diff
@@ -225,8 +227,33 @@ function addClutzAliases(
     // default is a keyword in typescript, so the name of the export being default means that it's a
     // default export.
     if (e.name === 'default') return false;
+
     // Otherwise check that some declaration is from the local module.
-    return e.declarations.some(d => d.getSourceFile() === origSourceFile);
+    for (const d of e.declarations) {
+      // This check helps us filter out export * rewritten exports. It appears,
+      // those point to the source file without any need to get through the
+      // getExportSpecifierLocalTargetSymbol API below.
+      if (d.getSourceFile() !== origSourceFile) {
```

Comment:
so, rephrasing, and maybe you can adjust the comments: if (1) a symbol is just a straight export, its declaration is in the original file, so the original check (this one) works. Then if (2) the export is renaming (`export {X as Y}`) then we want to consider this a local name. However if (3) it's an `import` followed by an `export` we don't want to do so.

Right?

I don't understand the reason behind (3) though. It looks to me like it's the same as (2) - sure, it's the same name, but in a different module, so it's the same as giving it a new name from that angle? I think TS allows you to reference an imported `X`:

```
import {X} from './foo';
export {X};
let y2: X;
```

Reply:
I think the renaming bits can be handled separately. The issue is purely about (3) vs pure reexport (without import).